support netcore3

jogibear9988 opened the pull request.

jogibear9988 commented:
you don't use a appveyor yml file atm. so I don't know how to fix build.

I use this in my appveyor yml:

image: Visual Studio 2019 Preview

install:
  - cmd: choco install dotnetcore-sdk --pre

jogibear9988 commented:
And if youd like to use csproj for the nuget build :

dotnet_csproj:
  patch: true
  file: '**\*.csproj'
  version: '{version}'
  package_version: '{version}' 
  assembly_version: '{version}'
  file_version: '{version}'
  informational_version: '{version}'

jogibear9988 commented:
but for the csproj patch to work, the version xml elements have to be in the csproj files (i've not added them atm)

Dirkster99 commented:
The yml file is now part of the master branch:
https://github.com/Dirkster99/AvalonDock/blob/master/appveyor.yml

jogibear9988 commented:
now maybe it should work.

you can adjust the version in the appveyor.yml

and if you'd like enable nuget publish

Dirkster99 commented:
Cool this works with Appveyor :-) I was actually looking for this integration for some time but found no hint about the choco trick - thanks for that. I am currently installing and will be looking at the branch this weekend, and hopefully, sometime next week we can get this into the master branch out of the door :-)

jogibear9988 commented:
I've removed the AssemblyVersion Attributes from the csharp files, cause I added the Version to csproj.
But I added only 1.00 so the Version is only correct on the Appveyor build

jogibear9988 commented:
The Nuget's should now also be added to the artifacts with my last commit.

So now you can decide, add them manualy or upload them directly by changeing appveyor.yml

jogibear9988 commented:
ok, now i'm finished.

nugets work, demo app works in netcore3

done.

Dirkster99 commented:
Yep. I just gave it some tests and its looking really good :-)

So, if I increase the version to say 3.5.12 I am setting this in the tag of the csproj file (right?):

```XML
        <AssemblyVersion>1.0.0.0</AssemblyVersion>
        <FileVersion>1.0.0.0</FileVersion>
```

I saw that you ported TestApp.

I would also like to port the other demo projects but to do this I have to port all dependent control projects first, right? I guess, I'll look at that as soon as this is on master which will be very soon thanx to your cool PR :-)

jogibear9988 commented:
yes, but at the moment appveyor oberwrites the csproj version.
so best would be to change in appveyor json and use packages from artifacts page, or let appveyor upload them

jogibear9988 commented:
I think netcore could also use some wpf packages directly, but dont know why
